dynawo/sources/Modeler/Common/DYNSubModelFactory.cpp

```diff
@@ -49,19 +49,31 @@ SubModelFactories::~SubModelFactories() {
 }
 
 SubModelFactories::SubmodelFactoryIterator SubModelFactories::find(const std::string& lib) {
+#ifdef LANG_CXX11
```

Comment:
why a mutex in a  read only method?

Reply:
In case we perform a get and a set at the same time